app/controllers/lectures_controller.rb

```diff
@@ -300,17 +300,19 @@ def check_for_subscribe
   end
 
   def lecture_params
-    params.require(:lecture).permit(:course_id, :term_id, :teacher_id,
-                                    :start_chapter, :absolute_numbering,
-                                    :start_section, :organizational, :locale,
-                                    :organizational_concept, :muesli,
-                                    :organizational_on_top,
-                                    :disable_teacher_display,
-                                    :content_mode, :passphrase, :sort,
-                                    :comments_disabled,
-                                    :submission_max_team_size,
-                                    :submission_grace_period,
-                                    editor_ids: [])
+    allowed_params = [:term_id, :start_chapter, :absolute_numbering,
+                      :start_section, :organizational, :locale,
+                      :organizational_concept, :muesli, :organizational_on_top,
+                      :disable_teacher_display, :content_mode, :passphrase,
+                      :sort, :comments_disabled, :submission_max_team_size,
+                      :submission_grace_period]
+    if action_name == 'update' && current_user.can_update_personell?(@lecture)
+      allowed_params.concat([:teacher_id, editor_ids: []])
```

Comment:
I don't understand the syntax here. Why is `editor_ids` not preprended by `:` in this case? Is `[]` the default argument for the parameter?

Reply:
This is because `editor_ids` is supposed to be an Array. Therefore the `editors_ids` key is mapped to `[]`, see [here](https://edgeguides.rubyonrails.org/action_controller_overview.html#permitted-scalar-values).